### ISSUE TITLE
[Bug] Wrong type generated in Map

### ISSUE THREAD

**Issue by dosomder** [jqno/equalsverifier#1022]
### Describe the bug

equalsverifier fills maps with the wrong type. In the example code you can see `map2` has keys with type String instead of Integer. It seems that if you have more than one map in the class, it uses the key / value type of the first map for all.
This can lead to issues if you have some logic in the equals method.

With 3.16.2 this problem did not happen.

### Steps to reproduce

Run the test below

### Error message and version number

EqualsVerifier 3.17.3, JDK 21.0.5 running on classpath, on Linux

### Code: EqualsVerifier invocation

```
public class TestClassTest {
  @Test
  void testEquals() {
    EqualsVerifier.forClass(TestClass.class).usingGetClass().verify();
  }
}
```

### Code: class under test

```
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class TestClass {
  private final Map<String, String> map1;
  private final Map<Integer, String> map2;

  public TestClass() {
    this.map1 = new HashMap<>();
    this.map2 = new HashMap<>();
  }

  public Map<String, String> getMap1() {
    return this.map1;
  }

  public Map<Integer, String> getMap2() {
    return this.map2;
  }

  @Override
  public boolean equals(Object obj) {
    if (this.map2 != null) {
      for (Object key : this.map2.keySet()) {
        if (!(key instanceof Integer)) {
          throw new IllegalStateException("Key is %s instead of Integer".formatted(key.getClass().getSimpleName()));
        }
      }
    }

    if (this == obj) {
      return true;
    }
    if (obj == null || this.getClass() != obj.getClass()) {
      return false;
    }
    TestClass other = (TestClass) obj;
    return Objects.equals(this.map1, other.map1)
        && Objects.equals(this.map2, other.map2);
  }

  @Override
  public int hashCode() {
    return Objects.hash(this.map1, this.map2);
  }

  @Override
  public String toString() {
    return "{" +
        "\"class\": \"TestClass\"," +
        "\"map1\": " + this.getMap1() + "," +
        "\"map2\": " + this.getMap2() +
        "}";
  }
}
```

**Comment by jqno** [comment]
Thanks for telling me about this. I've reproduced the problem. It's a regression after a big refactoring that I did in 3.17.2. I hope to have the fix ready soon.

**Comment by jqno** [comment]
I've released version 3.17.4, it should solve your problem
